### PR TITLE
fix #384

### LINKED ISSUE #384: Support for evenodd fill rule
Support [evenodd](http://www.whatwg.org/specs/web-apps/current-work/multipage/the-canvas-element.html#dom-context-2d-fillrule-evenodd) fill rule.

Example:

```
var canvas = document.createElement('canvas');
var ctx = canvas.getContext('2d');
ctx.rect(1, 1, 50, 50);
ctx.rect(5, 5, 41, 41);
ctx.fill('evenodd');
```

> The "evenodd" value indicates the even-odd rule, wherein a point is considered to be outside a shape if the number of times a half-infinite straight line drawn from that point crosses the shape's path is even.

### PR DESCRIPTION
**Pull request by wsw0108**
can be verified by run:
`node examples/fill-evenodd.js`

### CODE REVIEW

**Comment by FuZhenn** [comment]
Great.

This pull request implements even-odd fill rule: #384 

```javascript
context.fill("evenodd")
```

File: examples/fill-evenodd.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+var fs = require('fs');
+
+var canvas = new Canvas(100, 100);
+var ctx = canvas.getContext('2d');
+
+ctx.fillStyle = '#f00';
+ctx.rect(0, 0, 100, 50);
+ctx.arc(50, 50, 50, 0, 2 * Math.PI);
+ctx.fill('evenodd');
+//ctx.fill();
```
Comment (LinusU) [inline examples/fill-evenodd.js]: please remove this line

File: examples/fill-evenodd.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (LinusU) [inline examples/fill-evenodd.js]: Please remove the empty line here

File: src/CanvasRenderingContext2d.cc [hunk 3]
```diff
@@ -1681,7 +1682,15 @@ NAN_METHOD(Context2d::Clip) {
 
 NAN_METHOD(Context2d::Fill) {
   Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  context->fill(true);
+  cairo_fill_rule_t rule = CAIRO_FILL_RULE_WINDING;
+  if (info.Length() == 1 && info[0]->IsString()) {
+      String::Utf8Value str(info[0]);
+      const char *c_arg = *str;
+      if (std::strcmp(c_arg, "evenodd") == 0) {
```
Comment (LinusU) [inline src/CanvasRenderingContext2d.cc]: Could this be `if (std::strcmp(*str, "evenodd") == 0) {`?

**Comment by wsw0108** [comment]
@LinusU I have update the commit, please check it.

**Comment by FuZhenn** [comment]
@LinusU 
Not in a hurry, but please take some time to have a look of this PR.
We need the even-odd fill rule feature now.

**Comment by LinusU** [comment]
@wsw0108 Sorry for the delay, I've had a lot going on lately, do you mind opening the pull request again and I would be happy to merge it 👌

**Comment by wsw0108** [comment]
@LinusU because the branch that create this PR gone, I create a new PR #762 .
Please review it.

**Pull request closed without merging**